src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateCopy = { ...state };
+  const finalArray = [];
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      stateCopy = Object.assign(stateCopy, action.extraData);
+      finalArray.push(stateCopy);
```

Comment:
no need to `push` after each case. you can do it once at the end of each iteration

Reply:
and you should push the copy of stateCopy if you don't want to get an array with the same object on all positions